### PR TITLE
Skeleton Ember SPA inside OSF

### PR DESCRIPTION
**Pull request by abought**
## Purpose
Created to track/coordinate work on future OSF development. By design and intent, the build process and asset locations for this addon are being kept separate from the main OSF bundle.
 https://openscience.atlassian.net/browse/EOSF-88

No rush to merge. This can also be moved to a feature branch on request.

## Changes
- Add a skeleton ember app that consumes the `ember-osf` addon, and lives inside the OSF repo.
- Add a new setting, `USE_EMBER`, to act as a feature flag.
- Add a new demonstration route, `/ember-sample`

## Side effects
None expected, but we should monitor for unexpected interactions between the OSF and Ember build processes.

## How to use
1. Serve the flask app as normal
2. Go into the `website-ember` folder and run setup instructions as normal for an ember project. (you may need to change the path to the `ember-osf` addon in package.json, depending on your directory structure)
3. Run the following terminal command to build the ember assets with livereload:
`ember build --output-path ../website/static/public/ember --watch`
4. Visit `localhost:5000/ember-sample` to see the page in your browser.

## Notes
Ember has its own build process. I highly recommend that any pycharm users consuming this app mark several subfolders of `website-ember` as ignored:
- `node_modules`
- `bower_components`
- `tmp`

## TODO
- [x] Add navbar/footer to application route
- [x] Fix issue where page sometimes failed to load because of endless server requests to user endpoint.
  - [x] Verify fix in further testing
- [x] Add support for including CSS
  - [ ] And document this in errata for the ember-osf addon
- [x] Set up to use cookie auth
- [x] Configure i18n
- [x] Set up linters and code quality tools
  - [x] JSCS
  - [x] Template linting
- [ ] Connect build process as an invoke task (**current task does not work!!!**)
-  ~~Connect Travis CI to run ember tests~~ (do in future PR)
- [x] Allow individual pages to control the browser title bar ([EOSF-91](https://openscience.atlassian.net/browse/EOSF-91))
- [x] Get navbar login button working

### Deployment notes
- [ ] Cookie auth will need settings for staging server (initial dev provided only localhost config)

## Ticket
https://openscience.atlassian.net/browse/EOSF-88

### CODE REVIEW

File: tasks/__init__.py [hunk 1]
```diff
@@ -958,6 +958,22 @@ def assets(ctx, dev=False, watch=False, colors=False):
 
 
 @task
+def spa_assets(ctx, production=False, watch=False):
+    """Build static assets for the OSF Ember SPA"""
```
Comment (abought) [inline tasks/__init__.py]: Note to self: this doesn't work at present, particularly because the utility needs to run in a separate folder.

**Comment by samchrisinger** [comment]
@sloria I've done a quick pass and this looks to be in good shape. Can you review and create a feature branch for us when you have time?

**Comment by sloria** [comment]
Sure. I'll take a look at this today

**Comment by abought** [comment]
This PR was used as the basis for a feature branch:
 https://github.com/CenterForOpenScience/osf.io/tree/feature/osf-ember

Should I:
a) Close this PR, or
b) Keep it open as a place for initial review/comments
?

**Pull request closed without merging**